eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/TagManagerUtilities.java

```diff
@@ -0,0 +1,330 @@
+   * @param dmWebEwon the {@link DMWebEwon} to get the tag name for
+   * @param dmWebEwonTag the {@link DMWebEwonTag} to get the tag name for
+   * @return the corresponding tag provider tag name
+   * @since 1.0.0
+   */
+  public static String getTagNameForProviderFromDMWebEwon(
+      EwonConnectorSettings connectorSettings, DMWebEwon dmWebEwon, DMWebEwonTag dmWebEwonTag) {
+    // Get M2Web Ewon tag object
+    M2WebEwonEBDTag cachedM2WebEwonTag =
+        CacheManager.getCachedM2WebEwonTagMap(dmWebEwon.getName()).get(dmWebEwonTag.getName());
+
+    // Get tag name for provider
+    boolean isSystemTag = false;
+    String tagNameForProvider;
+    if (cachedM2WebEwonTag != null) {
+      tagNameForProvider =
+          getTagNameForProviderFromRawTagInfo(
+              connectorSettings,
+              dmWebEwon.getName(),
+              dmWebEwonTag.getName(),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupA()),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupB()),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupC()),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupD()),
+              isSystemTag);
+    } else {
+      // Show warnin if debug enabled or if tag group sorting is enabled
+      if (connectorSettings.isSortTagsByGroup() || connectorSettings.isDebugEnabled()) {
+        LOGGER.warn(
+            "Could not find cached M2Web Ewon tag for the corresponding DMWeb Ewon tag: {}. Tag"
+                + " group sorting cannot be applied (if enabled), falling back to standard tag"
+                + " name!",
+            dmWebEwon.getName() + "/" + dmWebEwonTag.getName());
+      }
+      boolean isInGroup = false;
+      tagNameForProvider =
+          getTagNameForProviderFromRawTagInfo(
+              connectorSettings,
+              dmWebEwon.getName(),
+              dmWebEwonTag.getName(),
+              isInGroup,
+              isInGroup,
+              isInGroup,
+              isInGroup,
+              isSystemTag);
+    }
+    return tagNameForProvider;
+  }
+
+  /**
+   * Builds a {@link HistoricalTagValue} object from the specified tag provider and tag data point
+   * parameters.
+   *
+   * @param tagProviderName the name of the tag provider
+   * @param tagName the name of the tag
+   * @param tagDataType the {@link DataType} of the tag
+   * @param tagDataQuality the {@link DataQuality} of the tag value
+   * @param tagValueTimestamp the timestamp of the tag value
+   * @param tagValue the value of the tag
+   * @return the {@link HistoricalTagValue} object
+   * @since 1.0.0
+   */
+  public static HistoricalTagValue buildHistoricalTagValue(
+      String tagProviderName,
+      String tagName,
+      DataType tagDataType,
+      DataQuality tagDataQuality,
+      Date tagValueTimestamp,
+      Object tagValue) {
+    TagPath tagPath = TagPathParser.parseSafe(tagProviderName, tagName);
```

Comment:
Both `...sqltags...TagPath` and `...sqltags...TagPathParser` were used in previous version of the connector and are deprecated. I did look a bit in to this when migrating the functionality, but it was not immediately obvious what the alternative would be. Additionally, the `BasicScanclassHistorySet` we use is not deprecated but requires `...sqltags...TagPath`s instead of the standard tag path objects.

Reply:
This may require further investigation to find and implement an alternative.